### ISSUE TITLE
nullable inference of lists not accurate in PCF

### ISSUE THREAD

**Issue by OneCyrus** [ChilliCream/hotchocolate#1600]
**Describe the bug**
when using lists in PCF with C# 8.0 nullable support, we get a wrong schema. this code
```
#nullable enable
public bool GetData(
  List<string>? services)
{ }
```

generates

`getData(services: [String]): Boolean`

**Expected behavior**
this should generate `getData(services: [String!]): Boolean`. string should be non nullable

**Desktop (please complete the following information):**
- .Net Core 3.1

**Comment by michaelstaib** [comment]
Do you have a small repro. So far we were unable to reproduce it.

**Comment by OneCyrus** [comment]
here is a simple one: https://github.com/OneCyrus/HotChocolateInferenceBug

**Comment by michaelstaib** [comment]
Really strange... we have those under test and it works as advertised. Will have to look what is different with your setup. Thanks for the repro :D

**Comment by michaelstaib** [comment]
I can verify your issue!

**Comment by michaelstaib** [comment]
I will look at this next week when I decide what we pack into 10.4.1

**Comment by michaelstaib** [comment]
This one is now fixed with #1638
